### PR TITLE
Provide access to `Organization.instance_owner` from `Context`

### PR DESCRIPTION
**Pull request by exbinary**
### What
Provides access to `Organization.instance_owner` from the `Context` object introduced in #844.

### Why
`instance_owner`, like `SystemSetting.current_settings` is a request-scoped global that is better represented as part of `context` than by a class accessor. `Context` is also easier to inject and therefore mock in specs.

### Outstanding Questions, Concerns and Other Notes
Note that i've named the accessor `host_organization` instead of sticking with `instance_owner`.

While `instance_owner` makes intuitive sense when hung on `Organization`, i felt that `context.instance_owner` isn't as clear. If the reader doesn't already know the term, the method name is a bit cryptic and doesn't give any indication of the type of object returned. Since this method is accessed in several places far from `Organization`, `context.host_organization` seemed more transparent.

The downside is that we now have two terms in the codebase that refer to the same thing, one within the `Organization` class and table, and another everywhere else. We could take this thinking a bit further and rename `Organization.instance_owner` to `.host_organization`, leaving only the table column name as `is_instance_owner`.

Definitely open to thoughts about this (it's easy enough to rename either way).

### Pre-Merge Checklist

- [x] Security & accessibility have been considered
- [x] Tests have been added, or an explanation has been given why the features cannot be tested
- [x] Documentation and comments have been added to the codebase where required
- [x] Entry added to CHANGELOG.md not necessary
- [ ] Outstanding questions and concerns have been resolved
- [ ] Any next steps have been turned into Issues or Discussions as appropriate

### CODE REVIEW

File: app/models/context.rb [hunk 1]
```diff
@@ -15,6 +20,8 @@ def can_admin?
 
   def to_h
     {
```
Comment (maebeale) [inline app/models/context.rb:22]: Shouldn't this have already been in here? Or, we weren't calling it before?

File: spec/models/context_spec.rb [hunk 2]
```diff
@@ -38,11 +38,11 @@
...
 
   describe 'to_h' do
-    let(:user) { double 'User' }
-    let(:system_settings) { double 'SystemSettings' }
+    let(:user) { instance_double User, 'admin_role?' => true }
+    let(:system_settings) { instance_double SystemSetting }
     let(:context) { Context.new user: user, system_settings: system_settings }
 
-    def method_with_keyword_args positional_arg, non_context_keyword_arg:, user:, system_settings:
+    def method_with_keyword_args positional_arg, non_context_keyword_arg:, user:, system_settings:, **_rest
```
Comment (maebeale) [inline spec/models/context_spec.rb:45]: Why is this needed?

File: spec/models/context_spec.rb [hunk 3]
```diff
@@ -58,7 +58,7 @@ def method_with_keyword_args positional_arg, non_context_keyword_arg:, user:, sy
     subject { context.can_admin? }
 
     context 'sysadmin' do
-      let(:user) { instance_double 'User', admin_role?: false, sys_admin_role?: true }
+      let(:user) { instance_double User, admin_role?: false, sys_admin_role?: true }
```
Comment (maebeale) [inline spec/models/context_spec.rb:61]: Do we want both `can_admin?` and `admin_role?`

**Comment by maebeale** [comment]
I'd be fine wth changing the db name too! :)

File: spec/models/context_spec.rb (hunk 3 above)
Comment (maebeale) [reply spec/models/context_spec.rb:61]: And, nice fix! Do we need an extra test to make sure this works (since we didn't know this was broken before)?

File: spec/models/context_spec.rb (hunk 2 above)
Comment (exbinary) [reply spec/models/context_spec.rb:45]: Great question! I didn't mention anything about this in the commit or PR -- it's a bit irrelevant and i forgot about it :).

This spec itself is a bit confusing IMO, and maybe not even necessary. I was experimenting with how to easily destructure `context` into method call args, for the cases where the method needs multiple context properties.

The destructuring works nicely with `.to_h`, but in order for it to be really resilient, the method has to accept a 'glob' arg to cover all the other context attributes it doesn't care about. In this case, adding the new `host_organization` attribute to `Context.to_h` broke this spec -- because `method_with_keyword_args` wasn't expecting it.

**Review by maebeale — APPROVED** [review]

**Comment by exbinary** [comment]
Renamed the `is_instance_owner` column to `is_host`. Please give it one more look @maebeale .

**Comment by maebeale** [comment]
(the PR text should change too, right?)

File: app/controllers/asks_controller.rb [hunk 4]
```diff
@@ -16,9 +16,9 @@ def create
     if submission.save
       EmailNewSubmission.run!(
         submission: submission,
-        user: current_user,
-        system_setting: context.system_settings
-        organization: Organization.instance_owner
+        user: context.user,
+        system_setting: context.system_settings,
```
Comment (maebeale) [inline app/controllers/asks_controller.rb:20]: 👍

**Review by maebeale — APPROVED** [review]
looks great! love having this all match now!!!
